File: VB/Controllers/HomeController.vb
Imports System.Web.Mvc

Namespace PopupControlAfterRequestMvc
	Public Class HomeController
		Inherits Controller

		Public Function Index() As ActionResult
			Return View()
		End Function

		Public Function OnLoad() As ActionResult
			Return View()
		End Function

		Public Function AfterJQueryAjax() As ActionResult
			Return View()
		End Function

		Public Function AfterSyncFormSubmit() As ActionResult
			Return View()
		End Function

		Public Function AfterAsyncAjaxFormSubmit() As ActionResult
			Return View()
		End Function

		<HttpPost>
		Public Function AjaxAction() As ActionResult
			Return Content("After jQuery Ajax")
		End Function

		<HttpPost, ValidateInput(False)>
		Public Function ActionMethodForSyncForm(ByVal Description As String) As ActionResult
			ViewData("showPopup") = True
			ViewData("content") = Description
			Return View("AfterSyncFormSubmit")
		End Function

		<HttpPost, ValidateInput(False)>
		Public Function ActionMethodForAsyncAjaxForm(ByVal Description As String) As ActionResult
			Return Content(Description)
		End Function
	End Class
End Namespace

File: CS/Controllers/HomeController.cs
using System.Web.Mvc;

namespace PopupControlAfterRequestMvc {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }

        public ActionResult OnLoad() {
            return View();
        }

        public ActionResult AfterJQueryAjax() {
            return View();
        }

        public ActionResult AfterSyncFormSubmit() {
            return View();
        }

        public ActionResult AfterAsyncAjaxFormSubmit() {
            return View();
        }

        [HttpPost]
        public ActionResult AjaxAction() {
            return Content("After jQuery Ajax");
        }

        [HttpPost]
        [ValidateInput(false)]
        public ActionResult ActionMethodForSyncForm(string Description) {
            ViewData["showPopup"] = true;
            ViewData["content"] = Description;
            return View("AfterSyncFormSubmit");
        }

        [HttpPost]
        [ValidateInput(false)]
        public ActionResult ActionMethodForAsyncAjaxForm(string Description) {
            return Content(Description);
        }
    }
}